packages/pwa-kit-runtime/src/utils/logger.js

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright (c) 2024, Salesforce, Inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+const LOG_LEVELS = ['debug', 'info', 'warn', 'error']
+const DEFAULT_LOG_LEVEL = 'info'
+const DEFAULT_LOG_FORMAT = 'JSON'
+
+const isServerSide = typeof window === 'undefined'
+
+const LOG_LEVEL = isServerSide
+    ? process.env.PWAKIT_LOG_LEVEL || DEFAULT_LOG_LEVEL
+    : DEFAULT_LOG_LEVEL
+
+const LOG_FORMAT = isServerSide
+    ? process.env.PWAKIT_LOG_FORMAT || DEFAULT_LOG_FORMAT
+    : DEFAULT_LOG_FORMAT
+
+/**
+ * The PWAKITLogger provides structured logging with different log levels.
+ *
+ */
+export class PWAKITLogger {
+    /**
+     * Creates an instance of PWAKITLogger.
+     * @param {string} logLevel - The log level to set for the logger.
+     * @param {string} [format=DEFAULT_LOG_FORMAT] - The format in which to print log messages. Can be 'JSON' or 'TEXT'.
+     * @param {string} packageName - The name of the package where the logger is used.
+     */
+    constructor(logLevel = LOG_LEVEL, format = DEFAULT_LOG_FORMAT, packageName = '') {
+        this.logLevel = LOG_LEVELS.includes(logLevel) ? logLevel : DEFAULT_LOG_LEVEL
```

Comment:
Should `DEFAULT_LOG_FORMAT` be `LOG_FORMAT` ?

Reply:
Good catch!